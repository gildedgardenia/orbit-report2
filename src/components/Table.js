import React from 'react';

export default function Table(props) {
   const satellites = props.satellites;
   if(satellites.length === 0) {
     return(
       <h2>No satellites here!</h2>
     );
   }
 
   return (
     <table>
     <thead>
       <tr>
         <th>Id</th>
         <th>Name</th>
         <th>Type</th>
         <th>Launch Date</th>
         <th>Orbit Type</th>
         <th>Operational</th>
       </tr>
       </thead>
       <tbody>
         {satellites.map((satellite, index) => (
           <tr key={index}>
             <td>{satellite.id}</td>
             <td>{satellite.name}</td>
             <td>{satellite.type}</td>
             <td>{satellite.launchDate}</td>
             <td>{satellite.orbitType}</td>
             <td style={satellite.operational ? {color: "green"} : {color: "red"}}>{satellite.operational.toString()}</td>
           </tr>
         ))}
       </tbody>
       </table>
   );
 }