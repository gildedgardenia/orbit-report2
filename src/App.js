import './App.css';
import data from './satellites.json';
import Form from './components/Form';
import Table from './components/Table';

function App() {
  const satelliteData = data.satellites;

  return (
    <div className="App">
      <h1>Orbit Report</h1>
      <Table satellites={satelliteData} />
      <Form satellites={satelliteData} />
    </div>
  );
}

export default App;
