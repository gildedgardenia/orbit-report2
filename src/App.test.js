import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import App from './App';
import data from './satellites.json';
import Table from './components/Table';

test('renders learn header', () => {
  render(<App />);
  const headerElement = screen.getByText("Orbit Report");
  expect(headerElement).toBeInTheDocument();
});

test('table component', () => {
  const testRenderer = TestRenderer.create(
    <Table satellites={data.satellites}/>
  );
  expect(testRenderer.toJSON()).toMatchSnapshot();
});






