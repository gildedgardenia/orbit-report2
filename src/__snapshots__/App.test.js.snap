// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table component 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Id
      </th>
      <th>
        Name
      </th>
      <th>
        Type
      </th>
      <th>
        Launch Date
      </th>
      <th>
        Orbit Type
      </th>
      <th>
        Operational
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        Sirius 1
      </td>
      <td>
        Communication
      </td>
      <td>
        2007-11-17
      </td>
      <td>
        LOW
      </td>
      <td
        style={
          Object {
            "color": "green",
          }
        }
      >
        true
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        Vanguard 1
      </td>
      <td>
        Probe
      </td>
      <td>
        1958-03-17
      </td>
      <td>
        HIGH
      </td>
      <td
        style={
          Object {
            "color": "red",
          }
        }
      >
        false
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        SNAP 10-A
      </td>
      <td>
        Space Station
      </td>
      <td>
        1965-11-01
      </td>
      <td>
        HIGH
      </td>
      <td
        style={
          Object {
            "color": "red",
          }
        }
      >
        false
      </td>
    </tr>
    <tr>
      <td>
        4
      </td>
      <td>
        Hubble Space Telescope
      </td>
      <td>
        Telescope
      </td>
      <td>
        1990-05-24
      </td>
      <td>
        LOW
      </td>
      <td
        style={
          Object {
            "color": "green",
          }
        }
      >
        true
      </td>
    </tr>
    <tr>
      <td>
        5
      </td>
      <td>
        Envisat
      </td>
      <td>
        Probe
      </td>
      <td>
        2002-03-01
      </td>
      <td>
        LOW
      </td>
      <td
        style={
          Object {
            "color": "red",
          }
        }
      >
        false
      </td>
    </tr>
    <tr>
      <td>
        6
      </td>
      <td>
        Thermal Blanket
      </td>
      <td>
        Space Debris
      </td>
      <td>
        N/A
      </td>
      <td>
        LOW
      </td>
      <td
        style={
          Object {
            "color": "red",
          }
        }
      >
        false
      </td>
    </tr>
    <tr>
      <td>
        7
      </td>
      <td>
        Morelos III
      </td>
      <td>
        Communication
      </td>
      <td>
        2012-12-19
      </td>
      <td>
        LOW
      </td>
      <td
        style={
          Object {
            "color": "green",
          }
        }
      >
        true
      </td>
    </tr>
    <tr>
      <td>
        8
      </td>
      <td>
        IIR-M
      </td>
      <td>
        Positioning
      </td>
      <td>
        2007-08-17
      </td>
      <td>
        LOW
      </td>
      <td
        style={
          Object {
            "color": "green",
          }
        }
      >
        true
      </td>
    </tr>
    <tr>
      <td>
        9
      </td>
      <td>
        ISS
      </td>
      <td>
        Space Station
      </td>
      <td>
        1998-11-20
      </td>
      <td>
        LOW
      </td>
      <td
        style={
          Object {
            "color": "green",
          }
        }
      >
        true
      </td>
    </tr>
  </tbody>
</table>
`;
